compiler/ir/tsl/tiled_strided_layout.py

```diff
@@ -161,7 +162,7 @@ def largest_common_contiguous_block(
             next_stride = next(
                 (
                     (dim, depth, stride_self)
-                    for dim, depth, stride_self in self
+                    for dim, depth, stride_self in self_strides
```

Comment:
This change along with `[x for x in self]` seems unnecessary?

Reply:
It is necessary because we now also search for dynamic strides.

In the static scenario, we search for a stride (for example 4), and then increment the stride (to 16).
In the next step, we search for stride 16, and then increment again if it is found.

However in the dynamic case, the stride is None, and remains None. So now, we must make sure we do find the same stride twice. Therefore, a hit is removed from this list `self_strides`, as you can see on line 176

Reply:
Oh oops actually I was just a bit confused. You are obviously using the `__iter__` method here :laughing: nevermind the previous comment